app/views/govuk_publishing_components/components/_table.html.erb

```diff
@@ -2,46 +2,63 @@
   caption ||= nil
   head ||= []
   rows ||= []
-  first_cell_is_header ||=false
+  first_cell_is_header ||= false
   caption_classes ||= nil
-  sortable ||=false
+  sortable ||= false
+  filterable ||= false
 %>
 
-<%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, caption, {
-    sortable: sortable,
-    caption_classes: caption_classes
-  }) do |t| %>
+<div <% if filterable %> data-module="table-filter" <% end %> >
+  <% if filterable %>
+    <div class="govuk-table-filter">
+      <%= render "govuk_publishing_components/components/input", {
```

Comment:
Have you thought about how this filter will work with a screen reader? I've just tested it in Voiceover and there doesn't seem to be any feedback to indicate what's happening when you type something into the box.

Reply:
Yes, I think an aria-live attribute could be added to the parent `div`.

```
<% if filterable %>
  <div data-module="table-filter" aria-live="polite">
  ...
```

Reply:
And maybe something to associate the search input with the table?

Reply:
I’ve been looking into this and we discussed it at the FE meeting yesterday as well. There’s a couple of things to be taken into consideration: 
- As far as I can see the only changes to the table that a screen reader (or VoiceOver anyway) responds to is content changes. So in this case not changing the display value of a table row or even adding/removing it from the DOM (surprisingly) will trigger that. But we are not changing the content of a table row here. 
- It may be considered anyway that wrapping an entire table in a live region would cause a potentially massive amount of content to be related to the user which is not really acceptable.

A similar implementation was done [here](https://www.gov.uk/foreign-travel-advice) but in this case we only update a count of the number of rows displayed as a result of user action which would satisfy both of the points above. There is no requirement here to report the number of rows so unless we add that or have that as a kind of visually hidden element I am struggling to know how to deal with this.

Reply:
I think the key thing is that when a user types into the input something should tell them that they've made an impact on the UI somehow. Otherwise a screen reader user could be typing and they have no idea if what they're doing is actually working.

I like the idea of some visually hidden text that updates with a count of matching rows, that seems better than reading aloud the whole table (which could take some time, and also not clearly immediately show what has changed).

Reply:
I’ve added an element here to display (to assistive technology) the number of rows that are shown. It’s pretty similar to the way this is done on the Travel Advice page.

Reply:
I think this is fine.

There are a few differences between VoiceOver, NVDA and JAWS here:
- JAWS is fine.
- NVDA and Chrome is fine
- NVDA and Firefox. Changes are not announced (which is the same behaviour seen on https://www.gov.uk/foreign-travel-advice). I'm not sure if that's an issue with NVDA and Firefox and `aria-live`
- VoiceOver also seems fine although sometimes the announcement is interrupted.

Reply:
Thanks for looking at that @jon-kirwan, especially on other Screen Readers. I only tested it with VoiceOver and I noticed it was a little bit inconsistent but it’s the same as the Travel Advice page so I guess that’s acceptable.